src/django_pg_migration_tools/management/commands/migrate_with_timeouts.py

```diff
@@ -54,13 +210,137 @@ def _validate(self) -> None:
+            # parser instance for calling argparse's validation.
+            if isinstance(f, datetime.timedelta):
+                # time fields are expressed in milisseconds and so are printed
+                # back to the MC caller as such to avoid confusion.
+                ms = int(f.total_seconds() * 1_000)
+                if ms < 0:
+                    raise ValueError(f"{ms} is not a positive value.")
+            elif f < 0:
+                raise ValueError(f"{f} is not a positive value.")
+
+        if (
+            self.statement_timeout_retry_min_wait is not None
+            and self.statement_timeout_retry_max_wait is not None
+        ) and (
+            self.statement_timeout_retry_min_wait
+            > self.statement_timeout_retry_max_wait
         ):
-            super().handle(*django_migrate_args, **django_migrate_kwargs)
+            raise ValueError(
+                "statement-timeout-retry-min-wait-in-ms cannot be greater than "
+                "statement-timeout-retry-max-wait-in-ms."
+            )
+
+        if (
+            self.lock_timeout_retry_min_wait is not None
+            and self.lock_timeout_retry_max_wait is not None
+        ) and (self.lock_timeout_retry_min_wait > self.lock_timeout_retry_max_wait):
+            raise ValueError(
+                "lock-timeout-retry-min-wait-in-ms cannot be greater than "
+                "lock-timeout-retry-max-wait-in-ms."
+            )
+
+    def _migrate(self, *django_migrate_args: Any, **django_migrate_kwargs: Any) -> None:
+        migration_applied: bool = False
+        while self._can_migrate() and (migration_applied is False):
+            try:
+                with timeouts.apply_timeouts(
+                    using=django_migrate_kwargs["database"],
+                    lock_timeout=self.lock_timeout,
+                    statement_timeout=self.statement_timeout,
+                ):
+                    super().handle(*django_migrate_args, **django_migrate_kwargs)
+                    migration_applied = True
+            except timeouts.DBLockTimeoutError as exc:
+                self.lock_timeouts += 1
+                self._wait(
+                    exp=self.lock_timeout_retry_exp,
+                    min_wait=self.lock_timeout_retry_min_wait,
+                    max_wait=self.lock_timeout_retry_max_wait,
+                    current_attempt=self.lock_timeouts,
+                )
+                self._attempt_callback(exc)
+            except timeouts.DBStatementTimeoutError as exc:
+                self.statement_timeouts += 1
+                self._wait(
+                    exp=self.statement_timeout_retry_exp,
+                    min_wait=self.statement_timeout_retry_min_wait,
+                    max_wait=self.statement_timeout_retry_max_wait,
+                    current_attempt=self.statement_timeouts,
+                )
+                self._attempt_callback(exc)
+
+        if migration_applied is False:
```

Comment:
Pet peeve: comparison of booleans.

Prefer: `if not migration_applied:`

(and also above)

Reply:
2449e41f0a76fcfd8f2a7fc69ba531c7c4d6571d